src/Runner.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -54,7 +54,7 @@ public RunnerSettings LoadSettings()
             Trace.Info(nameof(LoadSettings));
             if (!IsConfigured())
             {
-                throw new InvalidOperationException("Not configured. Run config.(sh/cmd) to configure the runner.");
```

Comment:
InvalidOperationException is mapped to resultCode = 2, which is retriable error. With the changes above, running `run.sh|cmd` on an unconfigured runner would get stuck in a loop.

I think the runner being unconfigured should not result in retriable error

Reply:
We might want to exam other exceptions to make sure they don't cause a loop.

Reply:
Perhaps I should remove the `retry on ERR 2` part of the helpers, it wasn't part of the `run.sh|cmd` cycle before, we could skip it for now to limit the scope of this PR.